### ISSUE TITLE
Design wireframes for Chat Bot

### ISSUE THREAD

**Issue by kb0304** [WideChat/Rocket.Chat#307]

**Comment by KashishJagyasi** [comment]
### I’ve kept the same UI as there’s a library already created for it. there were some major problems I found which I’ve tried to solve 


### The agent photo on the livechat widget doesn't strike much what it might be about unless someone clicks and sees what it actually is.

- **Changed the header because it was there was no way of going back to the website without losing the chat**

- **The dropdown  was misguiding and looked like swiping the chat tab down.** (I tested it on 4 users and 2 of them found t confusing)

- **The 3 line + avatar was adding clutter. I don't think there's any need for the phone number of every agent.**

![Live Chat](https://user-images.githubusercontent.com/47213420/85716301-f3237f80-b709-11ea-984d-6effd52b234c.png)
### I propose, the widget should clearly be named chat with us and put in the Right edge so that it doesn't interfere with the website text 
![Live Chat-1](https://user-images.githubusercontent.com/47213420/85716314-f6b70680-b709-11ea-8a61-97f3abea28e6.png)
### The widget can also be safely placed on the bottom right to be in the scope of the thumb
![Live Chat-2](https://user-images.githubusercontent.com/47213420/85716318-f6b70680-b709-11ea-81dd-bc815579122d.png)
### The livechat agent handoff

1. 
![Livechat- Handoff](https://user-images.githubusercontent.com/47213420/85716323-f7e83380-b709-11ea-9b6d-4621b4b68b4a.png)
2. 
![Livechat- Handoff-1](https://user-images.githubusercontent.com/47213420/85716324-f880ca00-b709-11ea-9cf7-d6f4ed275520.png)

### Livechat menu options with agent
![Livechat- menu options agent](https://user-images.githubusercontent.com/47213420/85716329-f9196080-b709-11ea-9e0e-e7055828d02b.png)

### Livechat menu options with bot
![Livechat- menu options with bot](https://user-images.githubusercontent.com/47213420/85716330-f9196080-b709-11ea-954b-1ea86a96b64b.png)
### Change Department request
![Livechat- Change department](https://user-images.githubusercontent.com/47213420/85716320-f7e83380-b709-11ea-8571-2bcb9f26024d.png)

### Quick reply ( version 1) 

1. 
![Livechat- Quick reply (carrossel)](https://user-images.githubusercontent.com/47213420/85716332-f9b1f700-b709-11ea-969d-5ddb059bba7b.png)
2. 
![Livechat- Quick reply After slecting options](https://user-images.githubusercontent.com/47213420/85716335-f9b1f700-b709-11ea-8e99-39d8a23078b5.png)

### Quick reply tap state

1.
 ![Livechat- Quick reply](https://user-images.githubusercontent.com/47213420/85716342-fae32400-b709-11ea-95e8-1a532647ea85.png)
2.
 ![Livechat- Quick reply tap state](https://user-images.githubusercontent.com/47213420/85716338-fa4a8d80-b709-11ea-9477-fa9c9dcb77ab.png)
3. 
![Livechat- Quick reply tap state-1](https://user-images.githubusercontent.com/47213420/85716339-fae32400-b709-11ea-8250-9f975e5b11d9.png)

### Version 2 will make more sense if the replies have to be long. Also, horizontal scroll will be a problem in the desktop view

### Reply
![Livechat- Reply](https://user-images.githubusercontent.com/47213420/85716344-fb7bba80-b709-11ea-8eb9-04a264f154ed.png)
![Livechat-divider new messages](https://user-images.githubusercontent.com/47213420/85716346-fc145100-b709-11ea-9a4a-4736322f52fa.png)
### Waiting for agent state
![Livechat-opened](https://user-images.githubusercontent.com/47213420/85716349-fcace780-b709-11ea-97d3-b55aa6588ccb.png)

**Comment by ear-dev** [comment]
@KashishJagyasi Is it possible to do a side by side comparison with the designs from upstream, so we can see any changes you are recommending? 

One comment I have is that personally I don't like all the square buttons.  What do you think?  Is there another nice button design, or do you think this is best?

**Comment by KashishJagyasi** [comment]
Yeah sure. I'll make a Doc with the comparison or post it here. 
Yeah.. I was kinda going with the circular buttons before I saw this but as I said I used the library that was available. But I'll see if I can do anything about it.
@ear-dev

**Comment by ear-dev** [comment]
Thanks @KashishJagyasi .  I don't think we should limit ourselves at this point, but instead make our best, most beautiful design suggestions!

**Comment by KashishJagyasi** [comment]
### Changed all the components to increase the padding and make them more rounded. 
Rounded components are required here because We're already using a bot and it makes the customer feel less familiar. Rounded corners give a soothing feel to the designs and they must be used wherever we're trying to build a rapport with the customers.

Also, customer support will be used mostly when people have complaints or wanna know something. Rounded components ensure that they feel a sense of geniality and avoid hesitation or frustration. 
![Frame 8 (2)](https://user-images.githubusercontent.com/47213420/86122743-f053d080-baf5-11ea-8902-137d2d2e0f11.png)

**I need some feedback on the Quick reply screens. The first version will be helpful for mobile sites but it will be a problem on the desktop as using a horizontal scroll is not a good practice. Also, the second version will be better if the options provided are long threads.**

![Frame 9 (2)](https://user-images.githubusercontent.com/47213420/86122750-f21d9400-baf5-11ea-929e-357629316a18.png)
You may refer to the previous comment for more details on why I changed some of the elements.
@ear-dev @kb0304

**Comment by ear-dev** [comment]
@KashishJagyasi I just sent an email to you and another designer friend of ours, Ona.  Maybe she can help with some comments about your wireframes.  

But also, did you show these to Ivan yet?  Do you think he will be receptive to look at them?

**Comment by KashishJagyasi** [comment]
No, didn't discuss it with Ivan yet. I'm not really sure how receptive he will be but he's kinda busy with other stuff so.. @ear-dev

**Comment by dlassalle** [comment]
On the bot side, the only thing i dont see here that we want in the bot are:
- "Print chat" button
- Character limit under or on the text input
- UI for when chat input is disabled while bot/agent are performing an action (like waiting to run a speedtest)

All of these currently dont exist on the client, but we are working on them :)

**Comment by KashishJagyasi** [comment]
> * Character limit under or on the text input

Just to confirm: By character limit, you mean the user will be restrained to write messages within the character limit right?

I'll Add the other things soon, thanks for the feedback :)
@dlassalle

**Comment by dlassalle** [comment]
Yep! here is an example implementation: https://github.com/RocketChat/Rocket.Chat.Livechat/pull/443

**Comment by KashishJagyasi** [comment]
Wireframes for Character limit
![character limit](https://user-images.githubusercontent.com/47213420/88195594-d6715d80-cc5d-11ea-9ecc-16185be91972.png)

Alternatives for options as suggested by ona.
![options alternative](https://user-images.githubusercontent.com/47213420/88195559-cf4a4f80-cc5d-11ea-8b3b-2a92cd156189.png)

Disabled text field UI
![Livechat- Quick reply](https://user-images.githubusercontent.com/47213420/88195602-d8d3b780-cc5d-11ea-8ff1-e2e5fb3c69a3.png)

Change deptt UI
![Livechat- Change department](https://user-images.githubusercontent.com/47213420/88196420-cc039380-cc5e-11ea-959b-0c7191f773f3.png)

@dlassalle
@ear-dev
